app/models/announcement.rb

```diff
@@ -1,32 +1,31 @@
 class Announcement < ActiveRecord::Base
-  attr_accessible :start_date, :end_date, :button, :link, :body
-  
+  attr_accessible :id, :start_date, :end_date, :button, :link, :body
+
   validates :button, presence: true
   validates :link, presence: true
   validates :body, presence: true
-  validates :start_date, presence: true, uniqueness: true
-  validates :end_date, presence: true, uniqueness: true
-  validate :overlapping_dates, on: :create
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :overlapping_dates
   validate :start_date_before_end_date
 
-  def date_range
-    start_date..end_date
-  end
-
   def overlapping_dates
-    announcements = Announcement.all
-    errors.add(:base, "Announcement conflicts with other announcements") if announcements.any? { |announcement| announcement.date_range.overlaps? date_range }
+    if start_date && end_date
+      Announcement.find do | d |
```

Comment:
No i mean the whole thing. Run a test that exercises that validation, and tail the test log.
If this is pure sql, then it is a good validation, however if it is instantiating every Announcement in the database as an ActiveModel class and calling their date attributes, then that is bad.